apiserver/common/networkingcommon/networkconfigapi.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// The networkconfigapi package implements the network config parts
+// common to machiner and provisioner interface
+
+package networkingcommon
+
+import (
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/stateenvirons"
+)
+
+type NetworkConfigAPI struct {
+	st           *state.State
+	getCanModify common.GetAuthFunc
+	getAuthFunc  common.GetAuthFunc
```

Comment:
I'm not sure why you have both getCanModify and getAuthFunc, you seem to only be setting and calling the former. Maybe just a copy&paste issue?